src/Tween.ts

```diff
@@ -360,10 +378,11 @@ export class Tween<T extends UnknownProps> {
 			this._onUpdateCallback(this._object, elapsed)
 		}
 
-		if (elapsed === 1) {
+		if (this._duration === 0 || elapsedTime >= this._duration) {
 			if (this._repeat > 0) {
+				const completeCount = Math.min(Math.trunc((elapsedTime - this._duration) / durationAndDelay) + 1, this._repeat)
 				if (isFinite(this._repeat)) {
-					this._repeat--
+					this._repeat -= completeCount
```

Comment:
[Down below](https://github.com/tweenjs/tween.js/pull/564/files#diff-6f8cd5c42458ee058628fa97ed5c5c4bf529a252a662dc150844217f273f2561R423), regarding the chained tweens, there is this:

```js
					this._chainedTweens[i].start(this._startTime + this._duration)
```

Does this need to be updated too? Should it now be only `this._chainedTweens[i].start(this._startTime)` because we already advanced _startTime to where the next repeat would have been?

Reply:
This chained tween logic happens on the _next_ update right? Should I wonder if it needs to happen immediately in the updated where _repeat > 0, f.e. when we run `this._repeat -= completeCount` and _repeat becomes 0.

Reply:
This is getting complicated. I think removing some complexity from here, and instead adding something like a Timeline feature would make this a lot easier to understand:

#647

Reply:
Another issue with repeat(N) is that we cannot make the time go backwards, because we lose the initial _startTime every time we increment it. Timeline would make it easy to go forwards/backward anywhere on the timeline.

Reply:
The use of "repeat" and "chain" together is complicated. I agree with the need for additional unit tests.